### PR TITLE
interface: raw-usb: Adding ttyACM ttyACA permissions

### PR DESCRIPTION
**Pull request by kubiko**
Many z-wave, zigbee serial devices come with device node /dev/ttyACM[0-9]
some modem devices come as /dev/ttyAMA[0-9]
Adding this rule to raw-usb so those devices can be used

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: interfaces/builtin/raw_usb.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ const rawusbConnectedPlugAppArmor = `
 
 # Allow access to all ttyUSB devices too
 /dev/ttyUSB[0-9]* rw,
+/dev/ttyACM[0-9]* rw,
```
Comment (bboozzoo) [inline interfaces/builtin/raw_usb.go]: Reminds me of working on some cheap ass STM32 based devices using reference USB driver from ST.

**Comment by mvo5** [comment]
Looks good, we probably want @jdstrand to double check though.

**Comment by kubiko** [comment]
syntax fixed
Yes
ttyACM are so so far any z-wave, zigbee devices I had change to work with
ttyAMA are USB modem devices

File: interfaces/builtin/avahi_observe.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ network netlink,
 
 # Allow access to daemon to create socket
 /{,var/}run/avahi-daemon/  w,
-/{,var/}run/avahi-daemon/{pid,socket} rw,
+/{,var/}run/avahi-daemon/{pid,socket} rwk,
```
Comment (zyga) [inline interfaces/builtin/avahi_observe.go]: Is this related?

**Comment by jdstrand** [comment]
"ttyACM are so so far any z-wave, zigbee devices I had change to work with
ttyAMA are USB modem devices"

I'm sorry, I'm having trouble parsing this. Are you saying that ttyAMA devices *are* a USB device that just happens to show up as ttyAMA?

Are you saying that z-wave, zigbee devices are *not* some form of USB devices?

**Comment by jdstrand** [comment]
> Are you saying that z-wave, zigbee devices are _not_ some form of USB devices?

Googling it seems that z-wave and zigbee serial devices *are* some form of USB device. Can you confirm?

File: interfaces/builtin/raw_usb.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ const rawusbConnectedPlugAppArmor = `
 /dev/bus/usb/[0-9][0-9][0-9]/[0-9][0-9][0-9] rw,
 
 # Allow access to all ttyUSB devices too
-/dev/ttyUSB[0-9]* rw,
+/dev/tty{USB|ACM|AMA}[0-9]* rw,
```
Comment (bboozzoo) [inline interfaces/builtin/raw_usb.go]: That's a different set of devices than the one that was here before the branch got force pushed to. IIRC ttyAMA is the ARM UART controller, so it's wired to the internal bus on the SoC rather than the USB.

Comment (kubiko) [reply interfaces/builtin/raw_usb.go]: Yes I updated branch as I was told by  @alfonsosanchezbeato that some USB modem devices come as ttyAMA. @alfonsosanchezbeato can you confirm please.

**Comment by kubiko** [comment]
@jdstrand correct, zigbee and z-wave are typically USB devices which expose serial interface to communicate with them using AT commands. Both come as ttyACM[0-9]

**Comment by kubiko** [comment]
@jdstrand I might have been wrong about ttyAMA devices. I personally did not see any ttyAMA devices. So removing ttyAMA
However ttyACM is legitimate, and how zigbee z-wave device show up.

File: interfaces/builtin/raw_usb.go (hunk 3 above)
Comment (alfonsosanchezbeato) [reply interfaces/builtin/raw_usb.go]: Right, ttyAMA* devices are UART ones, I think we can drop those, sorry for the noise.

**Comment by jdstrand** [comment]
Thanks for making the changes. +1

**Comment by jdstrand** [comment]
@bboozzoo - my suggested change will fix the parser error.

File: interfaces/builtin/raw_usb.go [hunk 4]
```diff
@@ -35,7 +35,7 @@ const rawusbConnectedPlugAppArmor = `
 /dev/bus/usb/[0-9][0-9][0-9]/[0-9][0-9][0-9] rw,
 
 # Allow access to all ttyUSB devices too
-/dev/ttyUSB[0-9]* rw,
+/dev/tty{USB|ACM}[0-9]* rw,
```
[the remark this answers is not in the document]
Comment (mvo5) [reply interfaces/builtin/raw_usb.go]: I fixed this typo and pushed the update to this branch.
